### PR TITLE
11866 cancel button

### PR DESCRIPTION
**Pull request by jpdimond**

### CODE REVIEW

File: app/helpers/operations_helper.rb [hunk 1]
```diff
@@ -45,6 +45,11 @@ def format_operations_field(operation, field)
           link_to(t("operation.result_link_text.#{operation.kind}"),
             rails_blob_path(operation.attachment, disposition: "attachment"))
         end
+      when :pending
+        if operation.status == :pending && can?(:destroy, Operation)
```
Comment (cooperka) [inline app/helpers/operations_helper.rb:49]: I believe you can do `can?(:destroy, operation)` (lowercase o) to DRY up this check for :pending

File: app/helpers/operations_helper.rb [hunk 2]
```diff
@@ -45,6 +46,11 @@ def format_operations_field(operation, field)
...
+          link_to(t("common.cancel"), operation_path(operation), method: :delete,
+            data: { confirm: t("operation.destroy_confirm") })
```
[the remark this answers is not in the document]
Comment (cooperka) [reply app/helpers/operations_helper.rb:51]: Please autofix :)

**Review by cooperka — COMMENTED** [review]
Thanks for all the complex pairing, server support, documentation, and back-and-forth to get to this point Jill! Just a few little things.

Please also fix existing `spec/models/ability/operation_ability_spec.rb#L40`

File: app/controllers/operations_controller.rb [hunk 3]
```diff
@@ -26,6 +26,11 @@ def show
   end
 
   def destroy
+    # first destroy delayed job
+    if @operation.provider_job_id.present? && Delayed::Job.exists?(@operation.provider_job_id)
+      Delayed::Job.find(@operation.provider_job_id).destroy
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/controllers/operations_controller.rb:31]: sure

File: app/helpers/operations_helper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/helpers/operations_helper.rb:51]: done

File: spec/features/operations/operations_panel_spec.rb [hunk 4]
```diff
@@ -52,4 +52,13 @@
       end
     end
   end
+
+  context "pending operations" do
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply spec/features/operations/operations_panel_spec.rb:56]: yup!

[the remark this answers is not in the document]
Comment (jpdimond) [reply spec/features/operations/operations_panel_spec.rb:56]: yup!

**Pull request merged**